src/wasm-interpreter.h

```diff
@@ -1338,31 +1339,15 @@ class ExpressionRunner : public OverriddenVisitor<SubType, Flow> {
       return flow;
     }
     NOTE_EVAL1(curr->event);
-    auto exn = std::make_unique<ExceptionPackage>();
-    exn->event = curr->event;
+    WasmException exn;
+    exn.event = curr->event;
     for (auto item : arguments) {
-      exn->values.push_back(item);
+      exn.values.push_back(item);
     }
-    throwException(Literal::makeExn(std::move(exn)));
+    throwException(exn);
     WASM_UNREACHABLE("throw");
   }
-  Flow visitRethrow(Rethrow* curr) {
-    // FIXME Update the implementation to match the new spec
-    NOTE_ENTER("Rethrow");
-    WASM_UNREACHABLE("unimp");
-    /*
-    Flow flow = visit(curr->exnref);
-    if (flow.breaking()) {
-      return flow;
-    }
-    const auto& value = flow.getSingleValue();
-    if (value.isNull()) {
-      trap("rethrow: argument is null");
-    }
-    throwException(value);
-    WASM_UNREACHABLE("rethrow");
-    */
-  }
+  Flow visitRethrow(Rethrow* curr) { WASM_UNREACHABLE("unimp"); }
```

Comment:
This was moved to `RuntimeExpressionRunner`, because not `rethrow` needs an access to the exception stack.